automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataLibrarySpec.scala

```diff
@@ -52,6 +53,45 @@ class DataLibrarySpec extends FreeSpec with WebBrowserSpec with UserFixtures wit
     }
   }
 
+  "For a dataset with facets" in withWebDriver { implicit driver =>
+    val curatorUser = UserPool.chooseCurator
+    implicit val authToken: AuthToken = curatorUser.makeAuthToken()
+    withWorkspace(namespace, "Facets", attributes = Some(WorkspaceData.tags)) { wsName =>
+      withCleanUp {
+        //Hard Coded titles in our UI code
+        val cohortPhenotypeIndicationSection = "Cohort Phenotype/Indication"
+        val experimentalStrategySection = "Experimental Strategy"
+        val projectNameSection = "Project Name"
+        val primaryDiseaseSiteSection = "Primary Disease Site"
+        val dataUseLimitationSection = "Data Use Limitation"
+
+        //replacing values in the basic library dataset
+        val data = LibraryData.metadataBasic + ("library:datasetName" -> wsName,
+          "library:indication" -> s"$wsName+1",
+          "library:datatype" -> Seq(s"$wsName+2"),
+          "library:projectName" -> s"$wsName+3",
+          "library:primaryDiseaseSite" -> s"$wsName+4",
+          "library:dataUseRestriction" -> s"$wsName+5")
+
+        api.library.setLibraryAttributes(namespace, wsName, data)
+        register cleanUp api.library.unpublishWorkspace(namespace, wsName)
+        api.library.publishWorkspace(namespace, wsName)
+
+        withSignIn(curatorUser) { _ =>
+          val page = new DataLibraryPage().open
+          page.hasDataset(wsName) shouldBe true
+
+          //Verifying results
+          val expected = Map(cohortPhenotypeIndicationSection -> s"$wsName+1",
+            experimentalStrategySection -> s"$wsName+2",
+            projectNameSection -> s"$wsName+3",
+            primaryDiseaseSiteSection -> s"$wsName+4",
+            dataUseLimitationSection -> s"$wsName+5")
+
+          expected.foreach { case (title, item) =>
+            val childElement = cssSelector(s"[data-test-id='$title-facet_section'] [data-test-id='$item-item']").findElement
```

Comment:
We should make a FacetArea component that can properly model this structure, then add that component to LibraryPage.